legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendTextDocumentService.java

```diff
@@ -95,56 +94,67 @@ class LegendTextDocumentService implements TextDocumentService
     @Override
     public void didOpen(DidOpenTextDocumentParams params)
     {
+        LegendServerGlobalState globalState = this.server.getGlobalState();
+        this.server.checkReady();
+        TextDocumentItem doc = params.getTextDocument();
+        String uri = doc.getUri();
+        LOGGER.debug("Opened {} (language id: {}, version: {})", uri, doc.getLanguageId(), doc.getVersion());
+        LegendServerDocumentState docState = globalState.getOrCreateDocState(uri);
+
         this.server.runPossiblyAsync(() ->
         {
-            LegendServerGlobalState globalState = this.server.getGlobalState();
-            this.server.checkReady();
-            TextDocumentItem doc = params.getTextDocument();
-            String uri = doc.getUri();
-            LOGGER.debug("Opened {} (language id: {}, version: {})", uri, doc.getLanguageId(), doc.getVersion());
-            LegendServerDocumentState docState = globalState.getOrCreateDocState(uri);
-            docState.open(doc.getVersion(), doc.getText());
+            if (docState.open(doc.getVersion(), doc.getText()))
+            {
+                globalState.clearProperties();
+            }
+
+            globalState.logInfo("Opened document: " + uri);
         });
     }
 
     @Override
     public void didChange(DidChangeTextDocumentParams params)
     {
-        this.server.runPossiblyAsync(() ->
+        LegendServerGlobalState globalState = this.server.getGlobalState();
+        this.server.checkReady();
```

Comment:
I don't think it's necessary to call checkReady here, as that's done in getGlobalState.